encord/dataset.py

```diff
@@ -132,6 +133,39 @@ def add_users(self, user_emails: List[str], user_role: DatasetUserRole) -> List[
         """
         return self._client.add_users(user_emails, user_role)
 
+    def list_groups(self) -> Iterable[DatasetGroup]:
+        """
+        List all groups that have access to a particular dataset
+        """
+        while True:
+            page = self._client.list_groups(self.dataset_hash)
+            yield from page.results
+            break
+
+    def add_group(self, group_param: DatasetGroupParam):
```

Comment:
The `DatasetGroupParam` is a detail of implementation of POST handling in our HTTP client; no point in exposing this to anyone.

This should take just the fields separately and construct the parameter internally.

Same goes for the `EncordClientDataset.add_group`; the `Parameter` thing should only be created inside of that, IMNSHO.